nginx_module/src/http_request.rs

```diff
@@ -81,6 +81,13 @@ impl<'a> HttpRequest<'a> {
         }
     }
 
+    pub fn get_main_config<'b, Config>(&self, module: &ngx_module_t) -> Option<&'b Config> {
```

Comment:
I think it's better to mark this whole function as unsafe, as it assigns an arbitrary lifetime and a decision whether this  is safe or not can only be taken at call site, not inside the function

Reply:
with both functions?
i.e. get_main_config() and get_loc_config()?
this may be tricky since some code already uses the later